### PR TITLE
[BUG] Fix pointer error with Split Transport and LED/RGB Matrix

### LINKED ISSUE #13218: [Bug] Compile error with USE_I2C and RGB_MATRIX_SPLIT
## Describe the Bug

1. config `RGB_MATRIX_ENABLE = yes` and `RGB_MATRIX_DRIVER = WS2812`
2. define `USE_I2C` and `RGB_MATRIX_SPLIT`
3. Compile error occurs.

## System Information

 - Keyboard:  Original
   - https://github.com/yoyo-real/qmk_firmware/tree/feature/YK-4
 - Operating system: 
   - Windows
 - AVR GCC version: 
   - avr-gcc.exe (GCC) 8.4.0
 - ARM GCC version: 
   - arm-none-eabi-gcc.exe (GCC) 10.1.0
 - QMK Firmware version:
   - 0.13.9

## Additional Context

```
Compiling: quantum/process_keycode/process_magic.c                                                  [OK]
Compiling: quantum/process_keycode/process_grave_esc.c                                              [OK]
Compiling: quantum/split_common/transport.c                                                        quantum/split_common/transport.c: In function 'transport_master':
quantum/split_common/transport.c:169:5: error: cannot convert to a pointer type
     i2c_writeReg(SLAVE_I2C_ADDRESS, I2C_RGB_MATRIX_START, (void *)rgb_matrix_config, sizeof(i2c_buffer->rgb_matrix), TIMEOUT);
     ^~~~~~~~~~~~
quantum/split_common/transport.c: In function 'transport_slave':
quantum/split_common/transport.c:225:5: error: cannot convert to a pointer type
     memcpy((void *)i2c_buffer->rgb_matrix, (void *)rgb_matrix_config, sizeof(i2c_buffer->rgb_matrix));
     ^~~~~~
quantum/split_common/transport.c:225:5: error: cannot convert to a pointer type
quantum/split_common/transport.c:225:5: error: null argument where non-null required (argument 1) [-Werror=nonnull]
quantum/split_common/transport.c:225:5: error: null argument where non-null required (argument 2) [-Werror=nonnull]
cc1.exe: all warnings being treated as errors
 [ERRORS]
 |
 |
 |
make[1]: *** [tmk_core/rules.mk:436: .build/obj_yk_4_default/quantum/split_common/transport.o] エラー 1
Make finished with errors
make: *** [Makefile:530: yk_4:default] エラー 1
```

### PR DESCRIPTION
**Pull request by drashna**
## Description

If i2c is enabled for the split transport, and RGB Matrix or LED Matrix is enabled, it cannot convert the pointer type, and errors out. 

Fortunately, both are unions, and we can use the "raw" property to work around this. 

Same change applied to serial, just to be safe.

## Types of Changes

- [x] Core
- [x] Bugfix

## Issues Fixed or Closed by This PR

* Fixes #13218

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: quantum/split_common/transport.c [hunk 1]
```diff
@@ -161,12 +161,12 @@ bool transport_master(matrix_row_t master_matrix[], matrix_row_t slave_matrix[])
 #    endif
 
 #    if defined(LED_MATRIX_ENABLE) && defined(LED_MATRIX_SPLIT)
-    i2c_writeReg(SLAVE_I2C_ADDRESS, I2C_LED_MATRIX_START, (void *)led_matrix_eeconfig, sizeof(i2c_buffer->led_matrix), TIMEOUT);
+    i2c_writeReg(SLAVE_I2C_ADDRESS, I2C_LED_MATRIX_START, (void *)led_matrix_eeconfig.raw, sizeof(i2c_buffer->led_matrix.raw), TIMEOUT);
```
Comment (tzarc) [inline quantum/split_common/transport.c:164]: Shouldn't it be this?
```suggestion
    i2c_writeReg(SLAVE_I2C_ADDRESS, I2C_LED_MATRIX_START, (void *)&led_matrix_eeconfig.raw, sizeof(i2c_buffer->led_matrix.raw), TIMEOUT);
```

File: quantum/split_common/transport.c [hunk 2]
```diff
@@ -161,12 +161,12 @@ bool transport_master(matrix_row_t master_matrix[], matrix_row_t slave_matrix[])
...
     bool suspend_state = led_matrix_get_suspend_state();
     i2c_writeReg(SLAVE_I2C_ADDRESS, I2C_LED_SUSPEND_START, (void *)suspend_state, sizeof(i2c_buffer->led_suspend_state), TIMEOUT);
 #    endif
 #    if defined(RGB_MATRIX_ENABLE) && defined(RGB_MATRIX_SPLIT)
-    i2c_writeReg(SLAVE_I2C_ADDRESS, I2C_RGB_MATRIX_START, (void *)rgb_matrix_config, sizeof(i2c_buffer->rgb_matrix), TIMEOUT);
+    i2c_writeReg(SLAVE_I2C_ADDRESS, I2C_RGB_MATRIX_START, (void *)rgb_matrix_config.raw, sizeof(i2c_buffer->rgb_matrix.raw), TIMEOUT);
```
Comment (tzarc) [inline quantum/split_common/transport.c:169]: Shouldn't it be this?
```suggestion
    i2c_writeReg(SLAVE_I2C_ADDRESS, I2C_RGB_MATRIX_START, (void *)&rgb_matrix_config.raw, sizeof(i2c_buffer->rgb_matrix.raw), TIMEOUT);
```

File: quantum/split_common/transport.c [hunk 3]
```diff
@@ -218,11 +218,11 @@ void transport_slave(matrix_row_t master_matrix[], matrix_row_t slave_matrix[])
 #    endif
 
 #    if defined(LED_MATRIX_ENABLE) && defined(LED_MATRIX_SPLIT)
-    memcpy((void *)i2c_buffer->led_matrix, (void *)led_matrix_eeconfig, sizeof(i2c_buffer->led_matrix));
+    memcpy((void *)i2c_buffer->led_matrix.raw, (void *)led_matrix_eeconfig.raw, sizeof(i2c_buffer->led_matrix.raw));
```
Comment (tzarc) [inline quantum/split_common/transport.c:221]: Shouldn't it be this?
```suggestion
    memcpy((void *)&i2c_buffer->led_matrix.raw, (void *)&led_matrix_eeconfig.raw, sizeof(i2c_buffer->led_matrix.raw));
```

File: quantum/split_common/transport.c [hunk 4]
```diff
@@ -218,11 +218,11 @@ void transport_slave(matrix_row_t master_matrix[], matrix_row_t slave_matrix[])
...
     led_matrix_set_suspend_state(i2c_buffer->led_suspend_state);
 #    endif
 #    if defined(RGB_MATRIX_ENABLE) && defined(RGB_MATRIX_SPLIT)
-    memcpy((void *)i2c_buffer->rgb_matrix, (void *)rgb_matrix_config, sizeof(i2c_buffer->rgb_matrix));
+    memcpy((void *)i2c_buffer->rgb_matrix.raw, (void *)rgb_matrix_config.raw, sizeof(i2c_buffer->rgb_matrix.raw));
```
Comment (tzarc) [inline quantum/split_common/transport.c:225]: Shouldn't it be this?
```suggestion
    memcpy((void *)&i2c_buffer->rgb_matrix.raw, (void *)&rgb_matrix_config.raw, sizeof(i2c_buffer->rgb_matrix.raw));
```

**Review by tzarc — CHANGES_REQUESTED** [review]

File: quantum/split_common/transport.c (hunk 1 above)
Comment (drashna) [reply quantum/split_common/transport.c:164]: Probably

Comment (drashna) [reply quantum/split_common/transport.c:164]: Shouldn't the suspend bool have the same treatment, too?

Comment (tzarc) [reply quantum/split_common/transport.c:164]: > 
> 
> Shouldn't the suspend bool have the same treatment, too?

Yes.

File: quantum/split_common/transport.c (hunk 3 above)
Comment (y-k-m) [reply quantum/split_common/transport.c:221]: I think, maybe, need to swap memcpy's src and dist.

File: quantum/split_common/transport.c (hunk 4 above)
Comment (y-k-m) [reply quantum/split_common/transport.c:225]: Maybe need swap too same.

File: quantum/split_common/transport.c (hunk 1 above)
Comment (drashna) [reply quantum/split_common/transport.c]: Converting them all to pointers fixes the compilation errors, and removes the need to use `.raw`.  However, leaving the serial implementation with .raw.

File: quantum/split_common/transport.c (hunk 3 above)
Comment (drashna) [reply quantum/split_common/transport.c]: yeah, you're right.

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Closing due to closeness to breaking changes, and lets avoid merge conflicts

**Pull request closed without merging**